Source/JavaScriptCore/runtime/JSMicrotask.cpp

```diff
@@ -79,14 +86,18 @@ void runJSMicrotask(JSGlobalObject* globalObject, MicrotaskIdentifier identifier
     if (UNLIKELY(handlerArguments.hasOverflowed()))
         return;
 
-    if (UNLIKELY(globalObject->hasDebugger()))
+    if (UNLIKELY(globalObject->hasDebugger())) {
         globalObject->debugger()->willRunMicrotask(globalObject, identifier);
+        scope.clearException();
+    }
 
     profiledCall(globalObject, ProfilingReason::Microtask, job, handlerCallData, jsUndefined(), handlerArguments);
     scope.clearException();
```

Comment:
Let's call the `clearExceptionIfNotTermination` lambda (here and every other place in this function) instead.

Reply:
Done.